wkcuber/convert_nifti.py

```diff
@@ -72,129 +106,200 @@ def to_target_datatype(data: np.ndarray, target_dtype) -> np.ndarray:
         )
     )
 
-    write_webknossos_metadata(
-        str(target_path),
-        source_nifti_path.stem,
-        scale=scale,
-        exact_bounding_box={
-            "topLeft": offset,
-            "width": size[0],
-            "height": size[1],
-            "depth": size[2],
-        },
-    )
-
 
 def convert_folder_nifti(
-    source_folder_path, target_path, color_path, segmentation_path, scale
+    source_folder_path,
+    target_path,
+    color_subpath,
+    segmentation_subpath,
+    scale,
+    use_orientation_header=False,
+    bbox_to_enforce=None,
+    write_tiff=False,
 ):
     paths = list(source_folder_path.rglob("**/*.nii"))
 
-    if color_path not in paths and color_path is not None:
-        logging.warning(
-            "Specified color file {} not in source path {}!".format(
-                color_path, source_folder_path
+    color_path = None
+    segmentation_path = None
+    if color_subpath is not None:
+        color_path = target_path / color_subpath
+        if color_path not in paths:
+            logging.warning(
+                "Specified color file {} not in source path {}!".format(
+                    color_path, source_folder_path
+                )
             )
-        )
 
-    if segmentation_path not in paths and segmentation_path is not None:
-        logging.warning(
-            "Specified segmentation_file file {} not in source path {}!".format(
-                segmentation_path, segmentation_path
+    if segmentation_subpath is not None:
+        segmentation_path = target_path / segmentation_subpath
+        if segmentation_path not in paths:
+            logging.warning(
+                "Specified segmentation_file file {} not in source path {}!".format(
+                    segmentation_path, segmentation_path
+                )
             )
-        )
 
     logging.info("Segmentation file will also use uint8 as a datatype.")
 
+    conversion_args = {
+        "scale": scale,
+        "write_tiff": write_tiff,
+        "bbox_to_enforce": bbox_to_enforce,
+        "use_orientation_header": use_orientation_header,
+    }
     for path in paths:
         if path == color_path:
-            convert_nifti(path, target_path, "color", "uint8", scale)
+            convert_nifti(path, target_path, "color", "uint8", **conversion_args)
         elif path == segmentation_path:
-            convert_nifti(path, target_path, "segmentation", "uint8", scale)
+            convert_nifti(path, target_path, "segmentation", "uint8", **conversion_args)
         else:
-            convert_nifti(path, target_path, path.stem, "uint8", scale)
+            convert_nifti(path, target_path, path.stem, "uint8", **conversion_args)
 
 
-if __name__ == "__main__":
+def main():
     args = create_parser().parse_args()
     setup_logging(args)
 
     source_path = Path(args.source_path)
 
+    bbox_to_enforce = None
+    if args.enforce_bounding_box is not None:
```

Comment:
bounding box CLI parsing seems like a common task. It might make sense to move it into wkcuber utils. See how `add_scale` and `parse_scale` are done. Just an idea.